src/Domain/CampaignService.php

```diff
@@ -392,6 +393,31 @@ private function cachedMatchFundsRemaining(Campaign $campaign): Money
         return Money::fromSerialized($cachedAmountArray);
     }
 
+    private function cachedMetaCampaignMatchFundsRemaining(MetaCampaign $metaCampaign): Money
+    {
+        $id = $metaCampaign->getId();
+        if ($id === null) {
+            return Money::zero(Currency::GBP);
+        }
+
+        $cachedAmountArray = $this->cache->get(
+            key: self::CAMPAIGN_MATCH_AMOUNT_AVAILABLE_PREFIX . (string)$id,
+            callback: function (ItemInterface $item) use ($metaCampaign): array {
+                $item->expiresAfter(120); // two minutes
+                $startTime = $this->clock->now();
+                $returnValue = $this->matchFundsRemainingService->getFundsRemainingForMetaCampaign($metaCampaign)->jsonSerialize();
+                $endTime = $this->clock->now();
+
+                $diffSeconds = $startTime->diff($endTime)->f;
+                $this->log->info("Getting getFundsRemaining for metaCampaign {$metaCampaign->getSalesforceId()} took " . (string) $diffSeconds . "s");
+
+                return $returnValue;
+            }
+        );
+
+        return Money::fromSerialized($cachedAmountArray);
+    }
```

Comment:
code generally is getting pretty repetitive, in two minds about whether it should be cleaned up during the port over from SF or only at the end or next time we need to modify some of it.